### PR TITLE
Rename AddOEmbedProvider() and Obsolete OEmbedProviders()

### PR DESCRIPTION
**Pull request by erikjanwestendorp**
Suggested improvements in comments: https://github.com/umbraco/Umbraco-CMS/pull/11691#issuecomment-979885262 implemented

### CODE REVIEW

File: src/Umbraco.Core/Media/EmbedProviders/OEmbedResponse.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@ namespace Umbraco.Cms.Core.Media.EmbedProviders
     /// Wrapper class for OEmbed response
     /// </summary>
     [DataContract]
+    // TODO(V10) : change base class to OEmbedProviderBase
```
Comment (mikecp) [inline src/Umbraco.Core/Media/EmbedProviders/OEmbedResponse.cs:10]: Is this correct? It does not seem to be related to EmbedProvicerBase?

**Comment by mikecp** [comment]
Thanks @erikjanwestendorp to get this PR ready so quickly 👍 

I've just a question about the class OEmberResponse (see comment above).

Cheers!

File: src/Umbraco.Core/Media/EmbedProviders/OEmbedResponse.cs (hunk 1 above)
Comment (erikjanwestendorp) [reply src/Umbraco.Core/Media/EmbedProviders/OEmbedResponse.cs]: No it's not correct indeed. Removed in my latest commit 👍

**Comment by erikjanwestendorp** [comment]
@mikecp Thanks for your review. Just updated the PR 😄

**Review by mikecp — APPROVED** [review]

**Comment by mikecp** [comment]
Thanks @erikjanwestendorp ,

All set to merge now 😅 Thanks a lot!!
